### PR TITLE
Add a metric to show the count of blocks to remove

### PR DESCRIPTION
**Pull request by codings-dan**
### What changes are proposed in this pull request?

Add a metric to obtain the number of data blocks that have not been cleaned up temporarily.

### Why are the changes needed?

When the `free` command is executed, since the behavior of free to clear data blocks is asynchronous, even if free returns successfully, there are still data blocks that have not been cleared temporarily. If the `load` operation is executed again at this time, it will be found that the load execution speed is very fast, which is inaccurate. By increasing this metric, we can monitor the completion progress of the free command to a certain extent. Since the collection of this metric may affect performance, I set a switch to turn it on only when needed.

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

**Comment by codings-dan** [comment]
It seems that the ci test error has noting to do whith this code change

```
Error: 9.523 [ERROR] Errors: 
Error: 9.524 [ERROR] alluxio.client.cli.job.DistributedCommandsCancelStatsTest.testAsyncPersistCancelStats
Error: 9.524 [ERROR]   Run 1: DistributedCommandsCancelStatsTest.testAsyncPersistCancelStats:156 » Timeout T...
Error: 9.524 [ERROR]   Run 2: DistributedCommandsCancelStatsTest.testAsyncPersistCancelStats:156 » Timeout T...
02:58:49.524 [INFO] 
Warning: 524 [WARNING] Flakes: 
Warning: 524 [WARNING] alluxio.client.cli.JournalToolTest.dumpSimpleEmbeddedJournal
Error: 9.524 [ERROR]   Run 1: JournalToolTest.dumpSimpleEmbeddedJournal » Runtime java.lang.RuntimeException...
02:58:49.525 [INFO]   Run 2: PASS
02:58:49.525 [INFO] 
02:58:49.525 [INFO] 
Error: 9.525 [ERROR] Tests run: 71, Failures: 0, Errors: 1, Skipped: 0, Flakes: 1
```

**Comment by codings-dan** [comment]
@ggezer Can you help review the code? Thanks!

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java [hunk 1]
```diff
@@ -303,6 +303,11 @@ private DefaultBlockMaster(MetricsMaster metricsMaster, CoreMasterContext master
 
     MetricsSystem.registerGaugeIfAbsent(MetricKey.MASTER_LOST_BLOCK_COUNT.getName(),
         this::getLostBlocksCount);
+    if (ServerConfiguration.global().getBoolean(
+        PropertyKey.MASTER_COUNT_TO_REMOVE_BLOCKS_ENABLE)) {
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java:307]: Adding this metric makes sense to me. And I don't think you need to use a configuration to turn it on/off. If we control all metrics this way the configuration will be too much.

The performance is a concern, I get it. You can use `MetricsSysmtem.registerCachedGaugeIfAbsent()`. If the cached gauge update interval is not what you want, you should change the method to set the cache time.

File: core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java [hunk 2]
```diff
@@ -1364,6 +1369,10 @@ public int getLostBlocksCount() {
     return mLostBlocks.size();
   }
 
+  private int getToRemoveBlockCount() {
+    return mWorkers.stream().map(MasterWorkerInfo::getToRemoveBlockCount).reduce(0, Integer::sum);
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java:1373]: Proper locking is required. You can follow the patter in `getCapacityBytes()`. Check the `MasterWorkerInfo` java doc too.

Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java:1373]: Locking should not hurt your performance as long as you use the cached gauge.

File: core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java [hunk 3]
```diff
@@ -664,7 +663,6 @@ public LockResource lockWorkerMeta(EnumSet<WorkerMetaLockSection> lockTypes, boo
    *
    * @return the count
    */
-  @VisibleForTesting
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java:667]: ?

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Added a batch of comments. PTAL, thanks.

File: core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java (hunk 3 above)
Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java:667]: hmm seems i forgot to mention this method needs locking too, will add to the comment

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 4]
```diff
@@ -352,6 +352,11 @@ public MetricKey build() {
           .setDescription("Count of lost unique blocks")
           .setMetricType(MetricType.GAUGE)
           .build();
+  public static final MetricKey MASTER_TO_REMOVE_BLOCK_COUNT =
+      new Builder("Master.ToRemoveBlockCount")
+          .setDescription("Count of blocks to remove")
```
Comment (jiacheliu3) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:357]:
```suggestion
          .setDescription("Count of block replicas to be removed from the workers. If 1 block is to be removed from 2 workers, 2 will be counted here.")
```

**Review by jiacheliu3 — APPROVED** [review]
LGTM except one minor comment. Thanks for the improvement!

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**